libdevice/sanitizer_utils.cpp

```diff
@@ -524,24 +604,59 @@ ASAN_REPORT_ERROR_BYTE(store, true, 16)
 ASAN_REPORT_ERROR_N(load, false)
 ASAN_REPORT_ERROR_N(store, true)
 
+///
+/// ASAN initialize shdadow memory of local memory
+///
+
+static __SYCL_CONSTANT__ const char __mem_set_shadow_local[] =
+    "[kernel] set_shadow_local(beg=%p, end=%p, val:%02X)\n";
+
 DEVICE_EXTERN_C_NOINLINE void
 __asan_set_shadow_local_memory(uptr ptr, size_t size,
                                size_t size_with_redzone) {
+  // Since ptr is aligned to ASAN_SHADOW_GRANULARITY,
```

Comment:
all workitems are repeating the same operations.
it might be likely better to use a single workitem to do the operations and add a barrier at the end.

Reply:
I guess use barrier will be slower?